cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java

```diff
@@ -62,16 +61,16 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
   }
 
   @Override
-  public Predicate<EntityId> createFilter(Principal principal) throws Exception {
-    Set<Privilege> privileges = listPrivileges(principal);
-    final Set<EntityId> allowedEntities = new HashSet<>();
-    for (Privilege privilege : privileges) {
-      allowedEntities.add(privilege.getEntity());
-    }
+  public Predicate<EntityId> createFilter(final Principal principal) throws Exception {
     return new Predicate<EntityId>() {
       @Override
       public boolean apply(EntityId entityId) {
-        return allowedEntities.contains(entityId);
+        try {
+          enforce(entityId, principal, new HashSet<>(Arrays.asList(Action.values())));
+          return true;
+        } catch (Exception e) {
```

Comment:
It can only throw either an UnauthorizedException or an ExecutionException from the loading cache's get which will be thrown if the load method throws an exception, the load method here can only throw UnauthorizedException. So, the behavior is correct but I agree it is not apparent. I will add a check to improve both safety and clarity.